docs/src/ref_impl/build_and_run.md

```diff
@@ -188,3 +188,75 @@ In the case of the Icicle Kit, Ghaf deployment consists of creating an SD image
      * You can directly flash a NixOS image to onboard an MMC card: dd if=./result/nixos.img of=/dev/<YOUR_MMC_DEVICE> bs=32M.
 
 For more information on how to access the MMC card as a USB disk, see [MPFS Icicle Kit User Guide](https://tinyurl.com/48wycdka).
+
+## Building Installer image for Ghaf image
+
+In case you want to have installer for your Ghaf configuration you can use `ghaf.installer` options.
+
+It expects user to provide list of all enabled installer modules - set of prompts that will ask required information for installation process from user. And actual script that will install system based on them.
+
+Also one of the nixos-generators modules must be passed to `ghaf.installer.imgModules` to get installer image of that type.
+
+Additional modules can be written for your configuration in regular bash like this:
+
+```nix
+ghaf.installer.installerModules.flushImage = {
+  requestCode = ''
+    lsblk
+    read -p "Device name [e.g. sda]: " DEVICE_NAME
+  '';
+  providedVariables = {
+    deviceName = "$DEVICE_NAME";
+  };
+};
+```
+
+It will generate derivation that will build installer image with installation script that will be formatted in the following way:
+
+1. Builtin welcome screen.
+2. List of prompts in exactly the same way as in `ghaf.installer.enabledModules`.
+3. Installation script.
+
+On the actual run of an image generated script will be runned automatically at startup.
+
+Example configuration that will just flush pre built image on desired drive:
+
+```nix
+{config, ...}: {
+  ghaf.installer = {
+    enable = true;
+    imgModules = [
+      nixos-generators.nixosModules.raw-efi
+    ];
+    enabledModules = ["flushImage"];
+    installerCode = ''
+      echo "Starting flushing..."
+      if sudo dd if=${config.system.build.${config.formatAttr}} of=/dev/${config.ghaf.installer.installerModules.flushImage.providedVariables.deviceName} conv=sync bs=4K status=progress; then
+          sync
+          echo "Flushing finished successfully!"
+          echo "Now you can detach installation device and reboot to ghaf."
+      else
+          echo "Some error occured during flushing process, exit code: $?."
+          exit
+      fi
+    '';
+  };
+}
+```
+
+To bulid installer image write:
+
+```
+nix build .#nixosConfiguration.<CONFIGURATION>.config.system.build.installer
```

Comment:
`s/nixosConfiguration/nixosConfigurations` - for anyone copy-pasting the command to build the installer they have to change both `<CONFIGURATION>` and fix the typo